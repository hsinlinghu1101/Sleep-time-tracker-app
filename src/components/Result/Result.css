.data{
    background-color: #10316b;
    border-color: #f2f7ff;
}
.message{
    font-family: 'Marcellus SC', serif;
}
.end{
    height: 50px;
}

.mesA{
  color:rgb(247, 81, 81);
}
.mesB{
    color: rgb(45, 231, 7);
}
.mesC{
    color: yellow;
}

.font{
    font-family: 'Roboto Mono', monospace;
    font-size: 18px;
}
.top{
    width:60%;
    padding-bottom: 30px;
    margin-bottom: 30px;
    border-width: thick;
    border-bottom: darkgray ;
    border-style: dotted;
    border-top: hidden;
    border-left: hidden;
    border-right: hidden;
}