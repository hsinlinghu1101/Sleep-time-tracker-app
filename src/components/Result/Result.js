import React from 'react'



export default function Result(props){
    
   
        return(
            <section>      
              <h2>'{props.date}' - You got '{props.hours}' hours of sleep </h2>
              <p>{props.message}</p>
              <button>Edit</button>
              <button>Delete</button>
            </section>
           
          
        )
    
}