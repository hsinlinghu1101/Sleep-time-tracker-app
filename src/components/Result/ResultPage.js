import React from 'react';
import Result from './Result';
import NavBar from '../NavBar/NavBar'
import { format } from 'date-fns';
import DataAPiService from '../../Services/data-api-service';


export default class ResultPage extends React.Component {
 static defaultProps={
     match:{
         params:{}
     }
 }  
 
state={
    userData:[]
}

    componentDidMount() {
        
    DataAPiService.getData()
         .then(userData=> this.setState({
            userData
         }))  
    }

    handleDeleteData =(dataId)=>{
        this.props.history.push(`/`)
    }



    render() {
        
        
        const {userData} = this.state
        
        
        return (
            <main role="main">
                <header role="banner">
                    <NavBar />
               <h1>Daily Sleep Hours</h1>

                </header>
                {userData.map(data => {
                    
                    let hours = (new Date(data.wakeup_time) - new Date(data.bed_time))/1000/60/60
                    let message;
                    let age;
                    console.log(data)
                    if(age === 1) {
                        if(hours < 8){
                            message=' You need more sleep ! '
                        }else if (hours >=8 && hours <=10){
                            message='You had enough sleep'
                        }else{
                            message='You sleep too much'
                        }
                    }else if(age === 2 ){
                        if(hours < 7){
                            message=' You need more sleep ! '
                        }else if (hours >=7 && hours <=9){
                            message='You had enough sleep'
                        }else{
                            message='You sleep too much'
                        }
                    }else{
                        if(hours < 7){
                            message=' You need more sleep ! '
                        }else if (hours >=7 && hours <=8){
                            message='You had enough sleep'
                        }else{
                            message='You sleep too much'
                        }
                    }
                    return( 
                    <Result 
                    key={data.id} 
                    id={data.id} 
                    date={format(data.data_created, 'Do MMM YYYY')} 
                    hours={hours} 
                    message={message}
                    onDeleteData={this.handleDeleteData} />
                )})}

            </main>
        )
    }
}
