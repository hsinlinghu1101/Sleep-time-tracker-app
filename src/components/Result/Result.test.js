import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter as Router} from 'react-router-dom';
import Result from './Result';
import renderer from 'react-test-renderer';

it('renders without crashing', ()=>{
    const div = document.createElement('div')

    ReactDOM.render(
        <Router>
        <Result />
        </Router>, div)
        ReactDOM.unmountComponentAtNode(div)
    
})
it('renders the UI as expected', () => {
    const tree = renderer
      .create(
          <Router><Result/></Router>    
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });