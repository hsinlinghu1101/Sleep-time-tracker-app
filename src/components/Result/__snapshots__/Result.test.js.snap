// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  className="data"
>
  <div
    className="dateTime"
  >
    <h2 />
    <h2>
       hours of sleep 
    </h2>
  </div>
  <p
    className="mesA font"
  />
  <p
    className="mesB font"
  />
  <p
    className="mesC font"
  />
  <button
    className="btn data"
    onClick={[Function]}
  >
    Remove
  </button>
</section>
`;
